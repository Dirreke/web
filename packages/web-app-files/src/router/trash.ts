import { RouteComponents } from './router'
import { RouteLocationNamedRaw, RouteRecordRaw } from 'vue-router'
import { createLocation, isLocationActiveDirector } from './utils'

type trashTypes = 'files-trash-generic'

export const createLocationTrash = (name: trashTypes, location = {}): RouteLocationNamedRaw =>
  createLocation(name, location)

export const locationTrashGeneric = createLocationTrash('files-trash-generic')

export const isLocationTrashActive = isLocationActiveDirector<trashTypes>(locationTrashGeneric)

export const buildRoutes = (components: RouteComponents): RouteRecordRaw[] => [
  {
    path: '/trash',
    component: components.App,
    children: [
      {
        name: locationTrashGeneric.name,
        path: ':driveAliasAndItem(.*)?',
        component: components.Spaces.DriveResolver,
        meta: {
          authContext: 'user',
          patchCleanPath: true
        }
      }
    ]
  }
]
