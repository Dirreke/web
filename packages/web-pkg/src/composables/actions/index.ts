export * from './helpers'
export * from './files'
export * from './spaces'
export * from './types'
