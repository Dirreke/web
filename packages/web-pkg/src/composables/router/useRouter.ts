import { Router } from 'vue-router'
import { useService } from '../service'

export const useRouter = (): Router => {
  const hack = (window as any).__HACK__router
  if (hack) {
    return hack
  }
  return useService('$router')
}
