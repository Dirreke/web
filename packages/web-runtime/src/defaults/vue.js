import Vue from 'vue'
import WebPlugin from '../plugins/web'
import Avatar from '../components/Avatar.vue'
import focusMixin from '../mixins/focusMixin'
import lifecycleMixin from '../mixins/lifecycleMixin'
import VueMeta from 'vue-meta'
import VueRouter from 'vue-router'
import Vuex from 'vuex'

Vue.use(Vuex)
Vue.use(VueRouter)
Vue.use(WebPlugin)
Vue.use(VueMeta, {
  refreshOnceOnNavigation: true
})

Vue.component('AvatarImage', Avatar)

Vue.mixin(focusMixin)
Vue.mixin(lifecycleMixin)

// externalize Vue - this is not the Vue instance but the class
window.Vue = Vue

export default Vue
