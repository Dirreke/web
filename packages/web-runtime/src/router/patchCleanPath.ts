import { RouteLocation, RouteLocationNormalizedLoaded, RouteLocationRaw, Router } from 'vue-router'
import get from 'lodash-es/get'

// type: patch
// temporary patch till we have upgraded web to the latest vue router which make this obsolete
// this takes care that routes like 'foo/bar/baz' which by default would be converted to 'foo%2Fbar%2Fbaz' stay as they are
// should immediately go away and be removed after finalizing the update
// to apply the patch to a route add meta.patchCleanPath = true to it
// to patch needs to be enabled on a route level, to do so add meta.patchCleanPath = true property to the route
// c.f. https://github.com/vuejs/router/issues/ 1638
export const patchRouter = (router: Router) => {
  const bindResolver = router.resolve.bind(router)
  const cleanPath = (route) =>
    [
      ['%2F', '/'],
      ['//', '/']
    ].reduce((path, rule) => path.replaceAll(rule[0], rule[1]), route || '')

  router.resolve = (
    raw: RouteLocationRaw,
    currentLocation?: RouteLocationNormalizedLoaded
  ): RouteLocation & {
    href: string
  } => {
    const bindResolve = bindResolver(raw, currentLocation)

    if (!get(bindResolve, 'meta.patchCleanPath', false)) {
      return bindResolve
    }

    return {
      ...bindResolve,
      href: cleanPath(bindResolve.href),
      path: cleanPath(bindResolve.path),
      fullPath: cleanPath(bindResolve.fullPath)
    }
  }

  return router
}
