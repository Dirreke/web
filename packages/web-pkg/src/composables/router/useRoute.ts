import { Ref } from 'vue'
import { useRouter } from './useRouter'
import { RouteLocationNormalizedLoaded } from 'vue-router'

export const useRoute = (): Ref<RouteLocationNormalizedLoaded> => {
  const router = useRouter()
  return useRouter().currentRoute
}
