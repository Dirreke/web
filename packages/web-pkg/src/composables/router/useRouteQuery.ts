import { computed, Ref, unref } from 'vue'
import { useRouter } from './useRouter'
import { QueryValue } from './types'

export const useRouteQuery = (name: string, defaultValue?: QueryValue): Ref<QueryValue> => {
  const router = useRouter()

  return computed({
    get() {
      return unref(router.currentRoute).query[name] || defaultValue
    },
    async set(v) {
      if (unref(router.currentRoute).query[name] === v) {
        return
      }
      await router.replace({
        query: {
          ...unref(router.currentRoute).query,
          [name]: v
        }
      })
    }
  })
}
