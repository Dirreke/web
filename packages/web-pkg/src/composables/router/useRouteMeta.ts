import { computed, Ref, unref } from '@vue/composition-api'
import { useRoute } from './useRoute'

export const useRouteMeta = (key: string, defaultValue?: string): Ref<string> => {
  const route = useRoute()
  return computed(() => unref(route).meta[key] || defaultValue)
}
