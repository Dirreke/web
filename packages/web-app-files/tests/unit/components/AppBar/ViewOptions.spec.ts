import ViewOptions from 'web-app-files/src/components/AppBar/ViewOptions.vue'
import { useRouteQueryPersisted } from 'web-pkg/src/composables/router'
import { ref } from 'vue'
import {
  createStore,
  defaultPlugins,
  shallowMount,
  defaultStoreMockOptions,
  defaultComponentMocks
} from 'web-test-helpers'

jest.mock('web-pkg/src/composables/router')
const selectors = {
  pageSizeSelect: 'oc-page-size-stub',
  hiddenFilesSwitch: '[data-testid="files-switch-hidden-files"]',
  fileExtensionsSwitch: '[data-testid="files-switch-files-extensions-files"]'
}

describe('ViewOptions component', () => {
  it('sets the correct initial files page limit', () => {
    const perPage = '100'
    const { wrapper } = getWrapper({ perPage })
    expect(wrapper.find(selectors.pageSizeSelect).props().selected).toBe(perPage)
    expect(wrapper).toMatchSnapshot()
  })
  it('toggles the setting to show/hide hidden files', () => {
    const { wrapper, storeOptions } = getWrapper()
    wrapper.find(selectors.hiddenFilesSwitch).vm.$emit('change', false)
    expect(storeOptions.modules.Files.mutations.SET_HIDDEN_FILES_VISIBILITY).toHaveBeenCalled()
  })
  it('toggles the setting to show/hide file extensions', () => {
    const { wrapper, storeOptions } = getWrapper()
    wrapper.find(selectors.fileExtensionsSwitch).vm.$emit('change', false)
    expect(storeOptions.modules.Files.mutations.SET_FILE_EXTENSIONS_VISIBILITY).toHaveBeenCalled()
  })
})

function getWrapper({ perPage = '100' } = {}) {
  jest.mocked(useRouteQueryPersisted).mockImplementation(() => ref(perPage))

  const storeOptions = { ...defaultStoreMockOptions }
  const store = createStore(storeOptions)
  const mocks = defaultComponentMocks()
  return {
    storeOptions,
    mocks,
    wrapper: shallowMount(ViewOptions, {
      global: {
        mocks,
        plugins: [...defaultPlugins(), store]
      }
    })
  }
}
