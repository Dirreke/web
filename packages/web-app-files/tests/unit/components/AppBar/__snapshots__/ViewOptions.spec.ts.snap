// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewOptions component sets the correct initial files page limit 1`] = `
<div>
  <oc-button-stub appearance="raw" arialabel="Display customization options of the files list" class="oc-my-s oc-p-xs" gapsize="medium" id="files-view-options-btn" justifycontent="center" size="medium" submit="button" type="button" variation="passive">
    <oc-icon-stub accessiblelabel="" color="" filltype="line" name="settings-3" size="medium" type="span" variation="passive"></oc-icon-stub>
  </oc-button-stub>
  <oc-drop-stub class="oc-width-auto" dropid="files-view-options-drop" mode="click" paddingsize="medium" position="bottom-start" toggle="#files-view-options-btn">
    <oc-list-stub>
      <li class="files-view-options-list-item oc-mb-m">
        <oc-switch-stub label="Show hidden files" labelid="oc-switch-label-1"></oc-switch-stub>
      </li>
      <li class="files-view-options-list-item oc-my-m">
        <oc-switch-stub label="Show file extensions" labelid="oc-switch-label-1"></oc-switch-stub>
      </li>
      <li class="files-view-options-list-item oc-mt-m">
        <oc-page-size-stub class="files-pagination-size" label="Items per page" options="100,500" selected="100" selectid="oc-page-size-3"></oc-page-size-stub>
      </li>
    </oc-list-stub>
  </oc-drop-stub>
</div>
`;
